sources/packages/web/src/components/common/ProgressBar.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <v-slider
+    v-model="currentStatus"
+    :ticks="progressStepLabels"
+    :max="maximumAllowedProgressValue"
+    step="1"
+    show-ticks="always"
+    tick-size="0"
+    track-color="readonly"
+    :track-fill-color="progressBarColor"
+    :thumb-size="initialStepSize"
+    :thumb-color="initialStepColor"
+    track-size="20"
+    readonly
+    :disabled="disabled"
+    class="progress-slider mt-n2"
+  >
+    <template #tick-label="{ tick, index }">
+      <span v-if="index === currentStatus" class="label-bold black-color"
+        >{{ tick.label }}
+        <v-icon
+          v-if="progressLabelIcon"
+          :icon="progressLabelIcon"
+          :size="20"
+          :color="progressLabelIconColor"
+          class="pl-4"
+      /></span>
+      <span class="label-value black-color" v-else>{{ tick.label }} </span>
+    </template>
+  </v-slider>
+</template>
+<script lang="ts">
+import { PropType, defineComponent, computed } from "vue";
+
+export default defineComponent({
+  props: {
+    progressStepLabels: {
+      type: Object as PropType<Record<number, string>>,
+      required: true,
+    },
+    progressBarValue: {
+      type: Number,
+      required: true,
+    },
+    progressBarColor: {
+      type: String,
+      required: true,
+    },
+    disabled: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    initialStepSize: {
+      type: Number,
+      required: false,
+      default: 0,
+    },
+    initialStepColor: {
+      type: String,
+      required: false,
+      default: "warning",
+    },
+    progressLabelIcon: {
+      type: String,
+      required: false,
+    },
+    progressLabelIconColor: {
+      type: String,
+      required: false,
+    },
+  },
+  setup(props) {
+    const currentStatus = computed(() => props.progressBarValue);
```

Comment:
Is there a reason for the `props.progressBarValue` not being consumed directly?

Reply:
Because the props are immutable and cannot be assigned to a v-model(2 way binded) directly.

![image](https://user-images.githubusercontent.com/54600590/215588971-4ae0c367-ceb3-444d-bdb5-aeb19ac2fbd2.png)